src/v/cloud_topics/level_one/compaction/meta.h

```diff
@@ -94,4 +102,11 @@ struct object_output_t {
     std::unique_ptr<staging_file> staging_file;
 };
 
+using cmp_t = std::function<bool(
+  const log_compaction_meta_ptr, const log_compaction_meta_ptr)>;
```

Comment:
nit: consider using `const ptr&` to prevent incrementing and decrementing the ref count for every comparison in a sort.